subsys/net/ip/ipv6.c

```diff
@@ -1308,7 +1308,10 @@ int net_ipv6_send_na(struct net_if *iface, const struct in6_addr *src,
 	net_buf_add(frag, sizeof(struct net_icmpv6_na_hdr) + llao_len);
 
 	na_hdr = net_icmpv6_get_na_hdr(pkt, &hdr);
-	NET_ASSERT_INFO(na_hdr, "Too short fragment for NA");
+	if (!na_hdr) {
+		NET_DBG("fragment too short for NA");
```

Comment:
These changes overall are definitely in the right direction. But I'd consider it too big a jump to go from NET_ERR() (as used by NET_ASSERT*()) to NET_DBG here and in other cases. I'd suggest we add flow control to handle these condition, but keep the original logging level (it can be adjusted on a case by case basis later).

Reply:
Good point. Let me take a look.